### PR TITLE
clang-tidy: readability-identifier-naming

### PR DESCRIPTION
**Pull request by derekargueta**
Description: `readability-identifier-naming` enables the ability to ensure that letter-casing follows the style guide. This starts with just enabling for function/method parameters, which should always be `lower_case`.
Risk Level: low
Testing: existing
Docs Changes: N/A
Release Notes: N/A

Signed-off-by: Derek Argueta <dereka@pinterest.com>

### CODE REVIEW

File: source/common/common/utility.h [hunk 1]
```diff
@@ -521,7 +521,7 @@ class WelfordStandardDeviation {
    * Accumulates a new value into the standard deviation.
    * @param newValue the new value
```
Comment (alyssawilk) [inline source/common/common/utility.h]: update comment?

File: source/extensions/filters/network/kafka/request_codec.h [hunk 2]
```diff
@@ -61,9 +61,9 @@ class RequestDecoder
    * @param parserResolver supported parser resolver.
```
Comment (alyssawilk) [inline source/extensions/filters/network/kafka/request_codec.h]: Ditto :-)

File: source/extensions/transport_sockets/tls/context_impl.h [hunk 3]
```diff
@@ -73,7 +73,7 @@ class ContextImpl : public virtual Envoy::Ssl::Context {
    * @param pattern the pattern to match against (*.example.com)
    * @return true if the san matches pattern
    */
-  static bool dNSNameMatch(const std::string& dnsName, const char* pattern);
```
Comment (alyssawilk) [inline source/extensions/transport_sockets/tls/context_impl.h]: Huh, this name was already out of date.  Mind fixing it as long as we're in here?

File: source/extensions/transport_sockets/tls/context_impl.cc [hunk 4]
```diff
@@ -626,16 +626,16 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return false;
 }
 
-bool ContextImpl::dNSNameMatch(const std::string& dNSName, const char* pattern) {
-  if (dNSName == pattern) {
+bool ContextImpl::dNSNameMatch(const std::string& d_ns_name, const char* pattern) {
```
Comment (zuercher) [inline source/extensions/transport_sockets/tls/context_impl.cc]: Just a drive-by, but I think this can be `dns_name`. I'm also not sure why the method is `dNSNameMatch` and not `dnsNameMatch` but that can wait for another day.

Comment (derekargueta) [reply source/extensions/transport_sockets/tls/context_impl.cc]: looks like carryover from a funky BoringSSL name: https://github.com/google/boringssl/blob/master/include/openssl/x509v3.h#L197

It might've been intentional to preserve it, but I don't think it adds much value and breaks style uniformity in Envoy so I went ahead and updated it while we're here.

**Comment by derekargueta** [comment]
hmm not sure what to do about
```
ERROR: /source/test/tools/config_load_check/BUILD:18:1: Linking of rule '//test/tools/config_load_check:config_load_check_tool' failed (Exit 34). Note: Remote connection/protocol failed with: execution failed java.io.IOException: writing file failed
	at com.google.devtools.build.lib.unix.NativePosixFiles.write(Native Method)
	at com.google.devtools.build.lib.unix.UnixFileSystem$NativeFileOutputStream.write(UnixFileSystem.java:494)
	at com.google.devtools.build.lib.remote.AbstractRemoteActionCache$LazyFileOutputStream.write(AbstractRemoteActionCache.java:916)
	at com.google.common.hash.HashingOutputStream.write(HashingOutputStream.java:58)
	at java.base/java.io.FilterOutputStream.write(Unknown Source)
	at com.google.protobuf.ByteString$LiteralByteString.writeTo(ByteString.java:1320)
	at com.google.devtools.build.lib.remote.GrpcRemoteCache$2.onNext(GrpcRemoteCache.java:340)
	at com.google.devtools.build.lib.remote.GrpcRemoteCache$2.onNext(GrpcRemoteCache.java:335)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onMessage(ClientCalls.java:421)
	at io.grpc.ForwardingClientCallListener.onMessage(ForwardingClientCallListener.java:33)
	at io.grpc.ForwardingClientCallListener.onMessage(ForwardingClientCallListener.java:33)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:519)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

main error being `execution failed java.io.IOException: writing file failed`
